From 278b3a675353cda065d19a9de3d402f156d597f3 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:53:46 -0400
Subject: [fs] ext4: pair trace_ext4_writepages & trace_ext4_writepages_result

Message-id: <1414058125-4183-73-git-send-email-lczerner@redhat.com>
Patchwork-id: 98180
O-Subject: [RHEL 7.1 PATCH v2 072/171] ext4: pair trace_ext4_writepages & trace_ext4_writepages_result
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Ming Lei <ming.lei@canonical.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit bbf023c74dcf380769aec00aedd706a266b7d9ef

 Pair the two trace events to make troubeshooting writepages
 easier, and it should be more convinient to write a simple script
 to parse the traces.

 Cc: linux-ext4@vger.kernel.org
 Cc: Jan Kara <jack@suse.cz>
 Signed-off-by: Ming Lei <ming.lei@canonical.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index 2d3b6ac..c124925 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -2420,16 +2420,15 @@ static int ext4_writepages(struct address_space *mapping,
   * because that could violate lock ordering on umount
   */
  if (!mapping->nrpages || !mapping_tagged(mapping, PAGECACHE_TAG_DIRTY))
-  return 0;
+  goto out_writepages;
 
  if (ext4_should_journal_data(inode)) {
   struct blk_plug plug;
-  int ret;
 
   blk_start_plug(&plug);
   ret = write_cache_pages(mapping, wbc, __writepage, mapping);
   blk_finish_plug(&plug);
-  return ret;
+  goto out_writepages;
  }
 
  /*
@@ -2442,8 +2441,10 @@ static int ext4_writepages(struct address_space *mapping,
   * *never* be called, so if that ever happens, we would want
   * the stack trace.
   */
- if (unlikely(sbi->s_mount_flags & EXT4_MF_FS_ABORTED))
-  return -EROFS;
+ if (unlikely(sbi->s_mount_flags & EXT4_MF_FS_ABORTED)) {
+  ret = -EROFS;
+  goto out_writepages;
+ }
 
  if (ext4_should_dioread_nolock(inode)) {
   /*
-- 
1.7.1